### PR TITLE
Added test for endless service

### PR DESCRIPTION
**Pull request by yzaccc**
Added test for endless service.

image

### CODE REVIEW

File: tests/service_test.go [hunk 1]
```diff
@@ -73,6 +75,55 @@ func TestBasicService(t *testing.T) { // nolint
...
+				Protocol: v1.ProtocolTCP,
+				Port:     endlessServicePort,
+			}}
+			var service kubernetes.ServiceBase = kubernetes.NewService(cs, clusterSvc)
+			if _, err := service.Create(); err != nil {
+				t.Fatal(err)
+			}
+
+			// wait for final status
+			_, err := service.WaitForEndpoint()
```
Comment (HanFa) [inline tests/service_test.go:95]: nit: maybe combine it into one line? 
```
if _, err := service.Create(); err != nil {
   t.Fatal()
}
```

File: entities/service.go [hunk 2]
```diff
@@ -13,12 +13,18 @@ const (
...
 	ClusterIP    = "clusteip"
 	NodePort     = "nodeport"
-	ExternalName = "externalname"
+	ServiceName  = "servicename"
 	LoadBalancer = "loadbalancer"
 
 	Allprotocols = "allprotocols"
 )
 
+type Service struct {
```
Comment (HanFa) [inline entities/service.go:22]: nit: maybe change to `ServiceTemplate` to a bit differentiate from k8s service object?

**Review by HanFa — APPROVED** [review]
LTGM thanks @yzaccc

File: entities/service.go [hunk 3]
```diff
@@ -36,6 +42,20 @@ func NewService(p *Pod) *v1.Service {
 	}
 }
 
+// NewService returns the service boilerplate based on service template
+func NewServiceFromTemplate(t Service) *v1.Service {
+	serviceID++
```
Comment (knabben) [inline entities/service.go:47]: these global variables will bite us back when we start to run things concurrently

Comment (HanFa) [reply entities/service.go:47]: generate random uuid instead?

**Comment by knabben** [comment]
:shipit:

File: entities/service.go (hunk 3 above)
Comment (yzaccc) [reply entities/service.go:47]: will put lock on it

File: entities/service.go (hunk 2 above)
Comment (yzaccc) [reply entities/service.go:22]: entities/service is analogy to entities/pod, as `Pod` is a struct, wanted to keep the consistency.

File: entities/service.go (hunk 3 above)
Comment (yzaccc) [reply entities/service.go]: done

**Pull request merged**